### PR TITLE
chore: bump devDependencies to run on node 16 & npm 8

### PR DESCRIPTION
**Pull request by timowolf**
**What kind of change does this PR introduce?** (check one with "x")

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,7 +76,7 @@
     "tslint-config-prettier": "^1.17.0",
     "tslint-config-security": "^1.16.0",
     "tslint-config-swimlane": "^5.0.0",
-    "typescript": "^4.3.5"
```
Comment (dr-itz) [inline package.json:79]: why roll back TS?

**Comment by timowolf** [comment]
@dr-itz sorry, this was in draft mode. This is only a quick fix for the ci pipeline. I am not sure if we use the GitHub actions right, but the versions changed on the fly. So this is a quick fix.

E.g. should should discuss for a larger project dev update here. Still tslint and other archived libs are used. So we should switch to eslint, ng13, ivy

File: package.json [hunk 2]
```diff
@@ -65,7 +65,7 @@
     "karma-jasmine": "~4.0.0",
     "karma-jasmine-html-reporter": "^1.5.0",
     "ng-packagr": "^12.1.2",
-    "node-sass": "^4.13.1",
+    "node-sass": "^7.0.0",
```
Comment (dr-itz) [inline package.json:68]: can we get rid of node-sass completely and use dart-sass, drop this completely as `sass` is already a dependency of the angular build tools and instead change the scripts in `package.json` from `node-sass` to `sass`?

Comment (timowolf) [reply package.json:68]: sure, I was trying to make a minimum update to be able to build with node 16 and npm 8. I think when changing sass etc. we could also update the overall project including eslint etc.

File: package.json (hunk 1 above)
Comment (timowolf) [reply package.json:79]: ➜  ngx-datatable git:(chore/update-node-npm) ✗ npm install
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: ngx-datatable@0.0.0
npm WARN Found: typescript@4.2.4
npm WARN node_modules/typescript
npm WARN   dev typescript@"^4.3.5" from the root project
npm WARN   6 more (@angular-devkit/build-angular, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer typescript@"~4.2.3 || ~4.3.2" from @angular-devkit/build-angular@12.1.4
npm WARN node_modules/@angular-devkit/build-angular
npm WARN   dev @angular-devkit/build-angular@"~12.1.4" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: ngx-datatable@0.0.0
npm WARN Found: typescript@4.2.4
npm WARN node_modules/typescript
npm WARN   dev typescript@"^4.3.5" from the root project
npm WARN   6 more (@angular-devkit/build-angular, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer typescript@">=4.2.3 <4.4" from @angular/compiler-cli@12.1.5
npm WARN node_modules/@angular/compiler-cli
npm WARN   dev @angular/compiler-cli@"~12.1.4" from the root project
npm WARN   3 more (@angular-devkit/build-angular, @ngtools/webpack, ng-packagr)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: ngx-datatable@0.0.0
npm WARN Found: typescript@4.2.4
npm WARN node_modules/typescript
npm WARN   dev typescript@"^4.3.5" from the root project
npm WARN   6 more (@angular-devkit/build-angular, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer typescript@">=4.2.3 <4.4" from @angular/compiler-cli@12.1.5
npm WARN node_modules/@angular/compiler-cli
npm WARN   dev @angular/compiler-cli@"~12.1.4" from the root project
npm WARN   3 more (@angular-devkit/build-angular, @ngtools/webpack, ng-packagr)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: ngx-datatable@0.0.0
npm WARN Found: typescript@4.2.4
npm WARN node_modules/typescript
npm WARN   dev typescript@"^4.3.5" from the root project
npm WARN   6 more (@angular-devkit/build-angular, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer typescript@">=4.2.3 <4.4" from @angular/compiler-cli@12.1.5
npm WARN node_modules/@angular/compiler-cli
npm WARN   dev @angular/compiler-cli@"~12.1.4" from the root project
npm WARN   3 more (@angular-devkit/build-angular, @ngtools/webpack, ng-packagr)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: ngx-datatable@0.0.0
npm WARN Found: typescript@4.2.4
npm WARN node_modules/typescript
npm WARN   dev typescript@"^4.3.5" from the root project
npm WARN   6 more (@angular-devkit/build-angular, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer typescript@"~4.2.3 || ~4.3.2" from ng-packagr@12.2.5
npm WARN node_modules/ng-packagr
npm WARN   dev ng-packagr@"^12.1.2" from the root project
npm WARN   1 more (@angular-devkit/build-angular)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: ngx-datatable@0.0.0
npm WARN Found: typescript@4.2.4
npm WARN node_modules/typescript
npm WARN   dev typescript@"^4.3.5" from the root project
npm WARN   6 more (@angular-devkit/build-angular, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer typescript@"~4.2.3 || ~4.3.2" from ng-packagr@12.2.5
npm WARN node_modules/ng-packagr
npm WARN   dev ng-packagr@"^12.1.2" from the root project
npm WARN   1 more (@angular-devkit/build-angular)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: ngx-datatable@0.0.0
npm WARN Found: typescript@4.2.4
npm WARN node_modules/typescript
npm WARN   dev typescript@"^4.3.5" from the root project
npm WARN   6 more (@angular-devkit/build-angular, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer typescript@"^2.9.1 || ^3.0.1" from tslint-config-security@1.16.0
npm WARN node_modules/tslint-config-security
npm WARN   dev tslint-config-security@"^1.16.0" from the root project
npm WARN   1 more (tslint-config-swimlane)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: ngx-datatable@0.0.0
npm WARN Found: typescript@4.2.4
npm WARN node_modules/typescript
npm WARN   dev typescript@"^4.3.5" from the root project
npm WARN   6 more (@angular-devkit/build-angular, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer typescript@"^2.9.1 || ^3.0.1" from tslint-config-security@1.16.0
npm WARN node_modules/tslint-config-security
npm WARN   dev tslint-config-security@"^1.16.0" from the root project
npm WARN   1 more (tslint-config-swimlane)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: ngx-datatable@0.0.0
npm WARN Found: typescript@4.2.4
npm WARN node_modules/typescript
npm WARN   dev typescript@"^4.3.5" from the root project
npm WARN   6 more (@angular-devkit/build-angular, ...)

**Comment by timowolf** [comment]
@dr-itz I reset the "typescript": "^4.3.5" so that we have minimal change that builds on the ci. We should do a project update within another MR then.

**Review by dr-itz — APPROVED** [review]

**Pull request merged**